lib/Experian/IDAuth.pm

```diff
@@ -454,49 +437,38 @@ sub _do_192_authentication {
     my $residence = $self->{residence};
 
     # check for 192 supported countries
-    unless ($self->valid_country($self->{residence})) {
-        warn "Invalid residence: " . $self->{client_id} . ", Residence $residence";
-        return;
-    }
+    croak "Invalid residence: " . $self->{client_id} . ", Residence $residence" unless ($self->valid_country($self->{residence}));
 
     if (!$force_recheck && $self->has_done_request) {
         $self->{result_as_xml} = $self->get_192_xml_report;
         return 1;
     }
 
     # No previous result so prepare a request
-    $self->_build_request
-        || die("Cannot build xml_request for [" . $self->{client_id} . "/$search_option]");
 
-    eval { $self->_send_request } || do {
-        my $err = $@ || '?';
-        warn "could not send pdf request: $err";
-        return;
-    };
+    try { $self->_build_request } catch { croak "Cannot build xml_request for [" . $self->{client_id} . "/$search_option]" };
+
+    try { $self->_send_request } catch { croak "could not send xml request: $_" };
```

Comment:
same comment as above? instead of warning, we die now in catch as well?

Reply:
It was silently failing and it wasn't very helpful for debugging. Now all those warnings are treated as errors and being sent to the compliance team for further review.